types/tx.go

```diff
@@ -31,18 +31,11 @@ type Txs []Tx
 
 // Hash returns the simple Merkle root hash of the transactions.
 func (txs Txs) Hash() []byte {
-	// Recursive impl.
-	// Copied from tendermint/crypto/merkle to avoid allocations
-	switch len(txs) {
-	case 0:
-		return nil
-	case 1:
-		return txs[0].Hash()
-	default:
-		left := Txs(txs[:(len(txs)+1)/2]).Hash()
-		right := Txs(txs[(len(txs)+1)/2:]).Hash()
-		return merkle.SimpleHashFromTwoHashes(left, right)
+	txBz := make([][]byte, len(txs))
```

Comment:
can't we pass txs to merkle.SimpleHashFromByteSlices()? + cast before `merkle.SimpleHashFromByteSlices(([][]byte)(txs))`

Reply:
When I try that, i get the following `cannot convert txs (type Txs) to type [][]byte`. I'm hoping to fix this having to be the case with #2603